runtime/dart_vm.cc

```diff
@@ -467,17 +467,19 @@ DartVM::DartVM(std::shared_ptr<const DartVMData> vm_data,
     // Send the earliest available timestamp in the application lifecycle to
     // timeline. The difference between this timestamp and the time we render
     // the very first frame gives us a good idea about Flutter's startup time.
-    // Use a duration event so about:tracing will consider this event when
-    // deciding the earliest event to use as time 0.
+    // Use an instant event because the call to Dart_TimelineGetMicros
+    // may behave differently before and after the Dart VM is initialized.
+    // As this call is immediately after initialization of the Dart VM,
+    // we are interested in only one timestamp.
     if (settings_.engine_start_timestamp.count()) {
```

Comment:
Are there any users of `engine_start_timestamp`? If not, can we get rid of them. Why can't the caller specify the startup time instead?

Reply:
As far as I can find, this was the only user. I am unsure what you mean by that. Caller to where?

Reply:
Internal codesearch suggests this is the only usage.

Reply:
To make sure we are clear, is your suggestion to remove the `engine_start_timestamp` now and continue with the effort of this PR?

Reply:
If no one is using the field in the settings struct, I suggest removing it altogether. If someone is using it, I suggest setting the field in the embedder you are working on. As it stands, this field will go unused and existing users (if any) will break.

Reply:
It is not being used otherwise but our Flutter Engine, but should we make any consideration for 3rd party forks or embedders that we do not know about but might use it? `settings_` is a private field so I would not expect there to be.

Reply:
We promise stability to third-party embedders via the embedder API. Since this is non-public API, we'd simply need to make sure none of our own embedders that don't use the embedder API (e.g. iOS/Android/Fuchsia and internal embedders) aren't broken. If no one's using it (which seems to be the case) then you can safely delete it.

Reply:
The Android shell assigns to this value but never uses it. Its only use is in generating this event